# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/PBackgroundIDBSharedTypes.ipdlh
# Commit: d6aa1c5bc7d9
# Full Hash: d6aa1c5bc7d96f6a9bc0a76508c50cc852cf3f2e
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-10 15:51:07
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Change definition of OpenCursorParams subtypes to allow reducing duplicated code r=ttung
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40955
# ==============================================================================

diff -r 6e46dad972c8 -r d6aa1c5bc7d9 dom/indexedDB/PBackgroundIDBSharedTypes.ipdlh
--- a/dom/indexedDB/PBackgroundIDBSharedTypes.ipdlh	Tue Sep 10 16:47:08 2019 +0300
+++ b/dom/indexedDB/PBackgroundIDBSharedTypes.ipdlh	Tue Sep 10 16:47:26 2019 +0300
@@ -124,36 +124,43 @@
   IndexMetadata[] indexes;
 };
 
-struct ObjectStoreOpenCursorParams
-{
-  int64_t objectStoreId;
-  SerializedKeyRange? optionalKeyRange;
-  Direction direction;
-};
-
-struct ObjectStoreOpenKeyCursorParams
+struct CommonOpenCursorParams
 {
   int64_t objectStoreId;
   SerializedKeyRange? optionalKeyRange;
   Direction direction;
 };
 
+struct ObjectStoreOpenCursorParams
+{
+  CommonOpenCursorParams commonParams;
+};
+
+struct ObjectStoreOpenKeyCursorParams
+{
+  CommonOpenCursorParams commonParams;
+};
+
+struct CommonIndexOpenCursorParams
+{
+  CommonOpenCursorParams commonParams;
+  int64_t indexId;
+};
+
 struct IndexOpenCursorParams
 {
-  int64_t objectStoreId;
-  int64_t indexId;
-  SerializedKeyRange? optionalKeyRange;
-  Direction direction;
+  CommonIndexOpenCursorParams commonIndexParams;
 };
 
 struct IndexOpenKeyCursorParams
 {
-  int64_t objectStoreId;
-  int64_t indexId;
-  SerializedKeyRange? optionalKeyRange;
-  Direction direction;
+  CommonIndexOpenCursorParams commonIndexParams;
 };
 
+// TODO: Actually, using a union here is not very nice, unless IPDL supported
+// struct inheritance. Alternatively, if IPDL supported enums, we could merge
+// the subtypes into one. Using a plain integer for discriminating the
+// subtypes would be too error-prone.
 union OpenCursorParams
 {
   ObjectStoreOpenCursorParams;
